### PR TITLE
Store SOAPAction in a session key

### LINKED ISSUE #1673: WebServiceListener SOAPAction
**Describe the issue**
I want to replace an application that receives SOAP messages. This application receives all messages on the same uri and uses the SOAPAction-header to route them to the correct logic. When using the WebServiceListener I do not have access to the SOAPAction (header in SOAP 1.1, part or the Content-Type header in SOAP 1.2), blocking routing based on action

**Reporter**
Ricardo

**To Reproduce**
1. Use the below configuration
2. Use SOAPUI to send a SOAP message with the SOAPAction-header to the listener
3. Open the message in Ladybug
4. No SOAPAction is available in the session keys

**Screenshots**
N/A

**Configuration**
```
<Module>
	<Adapter name="Test">
		<Receiver name="Test">
			<WebServiceListener name="Test"
				address="test" />
		</Receiver >
		<Pipeline firstPipe="Echo">
			<exits>
				<exit path="EXIT" state="success" />
			</exits>
			<pipe
				name="Echo" 
				className="nl.nn.adapterframework.pipes.EchoPipe">
				<forward name="success" path="EXIT" />
			</pipe>
		</Pipeline>
	</Adapter>
</Module>
```

**Input**
```
[Pipeline Test.zip](https://github.com/ibissource/iaf/files/6270325/Pipeline.Test.zip)
```

**Environment**
```
FF! 7.7-20210402.153336: frank4ozb-java
Running on LPZWNL000014943 using Apache Tomcat/9.0.39
Java Version: OpenJDK Runtime Environment (1.8.0_252-b09)
Heap size: 300M, total JVM memory: 492M
Free disk space: 79GB, total disk space: 237GB
Up since: 2021-04-07 10:54:33 (2m)
```

**Additional Environment**
 - DBMS: N/A
 - Browser: N/A

**Additional Context**
Add any other context about the problem here. (f.e. ladybug report / test adapter with larva test)

### PR DESCRIPTION
**Pull request by alisihab**
Closes #1673

### CODE REVIEW

**Review by gvanbrakel — CHANGES_REQUESTED** [review]
Could you add a unit test or iaf-test to prove the behavior?

**Review by gvanbrakel — APPROVED** [review]

**Review by nielsm5 — CHANGES_REQUESTED** [review]

File: core/src/main/java/nl/nn/adapterframework/http/cxf/SOAPProviderBase.java [hunk 1]
```diff
@@ -292,4 +307,18 @@ private XmlBuilder getMimeHeadersXml(Iterator<MimeHeader> mimeHeaders) {
 		}
 		return xmlMimeHeaders;
 	}
+	
+	private String findAction(int start, String contentType) {
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/http/cxf/SOAPProviderBase.java:311]: You could determine the start index in this method
`contentType.indexOf("action=")`

File: core/src/test/java/nl/nn/adapterframework/http/cxf/SoapProviderTest.java [hunk 2]
```diff
@@ -318,4 +312,54 @@ public void receiveAndSendMessageWithAttachmentsTest() throws Throwable {
...
+	}
+
+	@Test
+	public void noSoapActionInSessionKey() throws Throwable {
+		SOAPMessage request = createMessage("soapmsg1_2.xml");
+		webServiceContext.getMessageContext().put("Content-Type", "application/soap+xml; somethingelse");
+		SOAPProvider.invoke(request);
+		webServiceContext.getMessageContext().clear();
+		assertNull(SOAPProvider.getSession().get("SOAPAction"));
+	}
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/http/cxf/SoapProviderTest.java:364]: All these tests could also be simplified by just testing `findAction(contentType)` with all these different contentTypes.
I would much rather see a test with soap1.1 and a soap 1.2 header, and vice-versa.
Happy flow vs unhappy flow

File: test/src/main/configurations/HTTP/ConfigurationWebServiceListenerSender.xml [hunk 3]
```diff
@@ -313,4 +313,40 @@
...
 	</adapter>
+	
+	<Adapter name="HttpSoapSender" description="Send Http request to the TestSoapActionListener.">
+		<receiver className="nl.nn.adapterframework.receivers.GenericReceiver"
+			name="HttpSoapSender">
+			<listener className="nl.nn.adapterframework.receivers.JavaListener"
+				serviceName="ibis4test-HttpSoapSender" />
+		</receiver>
+		<Pipeline firstPipe="sendRequest">
+			<exits>
```
Comment (nielsm5) [inline test/src/main/configurations/HTTP/ConfigurationWebServiceListenerSender.xml:324]: I think now that we have junit tests these are no longer required. Maybe its better to remove them

**Review by nielsm5 — CHANGES_REQUESTED** [review]

**Review by nielsm5 — APPROVED** [review]

**Pull request merged**